Source/WebCore/rendering/RenderLayerFilters.cpp

```diff
@@ -218,11 +225,16 @@ void RenderLayerFilters::applyFilterEffect(GraphicsContext& destinationContext)
 {
     LOG_WITH_STREAM(Filters, stream << "\nRenderLayerFilters " << this << " applyFilterEffect");
 
-    ASSERT(inputContext());
-    inputContext()->restore();
-
-    FilterResults results;
-    destinationContext.drawFilteredImageBuffer(m_sourceImage.get(), m_filterRegion, *m_filter, results);
+    if (m_filter->filterMode() == FilterMode::Native) {
+        auto styles = m_filter->createStyles();
+        destinationContext.removeFilterEffectStyles(styles.size());
```

Comment:
This will be a set of `endTransparencyLayer` and `restore`. It's unfortunate that we have to create the styles against; this seems error prone.